### PR TITLE
Memory stressing

### PR DESCRIPTION
**Pull request by TsaiAnson**
Added:
1. TB can now stress memory as a resource
2. SSH clients are now closed when not used

Removed:
1. Unused max_resource_capacity.py

### CODE REVIEW

File: src/run_throttlebot.py [hunk 1]
```diff
@@ -65,6 +67,8 @@ def convert_percent_to_raw(mr, current_mr_allocation, weight_change=0):
         return  weighting_to_blkio(weight_change, current_mr_allocation)
     elif mr.resource == 'NET':
         return weighting_to_net_bandwidth(weight_change, current_mr_allocation)
+    elif mr.resource == 'MEMORY':
+        return weighting_to_memory(weight_change, current_mr_allocation, mr.instances[0])
```
Comment (mchang6137) [inline src/run_throttlebot.py]: Memory allocations should be changed for *all* instances of the service.

**Review by mchang6137 — COMMENTED** [review]
Two comments:
1.) Changes to memory should be uniformly applied to all instances of a service in a deployment
2.) You should be checking to see if there is sufficient memory in the machine before adding or removing from it. Did you confirm that this is implemented?

File: src/run_throttlebot.py (hunk 1 above)
Comment (TsaiAnson) [reply src/run_throttlebot.py]: Yes, so memory is allocated like the other services (they are applied to all instances). However, when weighting the memory allocation value, I am making an assumption that the containers of the same service are uniform (aka the idle memory usage of one container is the same for all). That is why I only take one instance into consideration when weighting the value.

Comment (mchang6137) [reply src/run_throttlebot.py]: I see. Still, please maintain the abstraction. You can choose to parse a single item of that deployment inside the weighting_to_memory class.

**Pull request merged**